Building with qt 6.5 fails

```
/build/source/src/scriptable/scriptableproxy.cpp: In member function 'QByteArray ScriptableProxy::callFunctionHelper(const QByteArray&)':
/build/source/src/scriptable/scriptableproxy.cpp:924:23: error: no match for 'operator=' (operand types are 'QGenericArgument' and 'QMetaMethodArgument')
  924 |             args[i] = Q_ARG(QVariant, value);
      |                       ^~~~~
/nix/store/q5fzqbgghqf8nr947h8rvn80smrlr4rg-qtbase-6.5.0-dev/include/QtCore/qobjectdefs.h:82:21: note: candidate: 'constexpr QGenericArgument& QGenericArgument::operator=(const QGenericArgument&)'
   82 | class Q_CORE_EXPORT QGenericArgument
      |                     ^~~~~~~~~~~~~~~~
/nix/store/q5fzqbgghqf8nr947h8rvn80smrlr4rg-qtbase-6.5.0-dev/include/QtCore/qobjectdefs.h:82:21: note:   no known conversion for argument 1 from 'QMetaMethodArgument' to 'const QGenericArgument&'
/nix/store/q5fzqbgghqf8nr947h8rvn80smrlr4rg-qtbase-6.5.0-dev/include/QtCore/qobjectdefs.h:82:21: note: candidate: 'constexpr QGenericArgument& QGenericArgument::operator=(QGenericArgument&&)'
/nix/store/q5fzqbgghqf8nr947h8rvn80smrlr4rg-qtbase-6.5.0-dev/include/QtCore/qobjectdefs.h:82:21: note:   no known conversion for argument 1 from 'QMetaMethodArgument' to 'QGenericArgument&&'
/build/source/src/scriptable/scriptableproxy.cpp:952:17: error: operands to '?:' have different types 'QGenericReturnArgument' and 'QMetaMethodReturnArgument'
  952 |                 ? QGenericReturnArgument(returnValue.typeName(), static_cast<void*>(returnValue.data()) )
      |                 ^
```

--- a/.github/workflows/build-linux.yml
+++ b/.github/workflows/build-linux.yml
@@ -23,12 +23,21 @@ jobs:
               -ftest-coverage
               -fprofile-abs-path
             compiler_package: g++
+            with_qt6: false
 
           - os: ubuntu-20.04
             buildname: Linux-Clang
             build_type: Debug
             compiler: clang++
             compiler_package: clang
+            with_qt6: false
+
+          - os: ubuntu-latest
+            buildname: Qt 6
+            build_type: Debug
+            compiler: g++
+            compiler_package: g++
+            with_qt6: true
 
     steps:
       - name: Checkout source code
@@ -39,6 +48,8 @@ jobs:
 
       - name: Install dependencies
         uses: nick-invision/retry@v2
+        env:
+          WITH_QT6: '${{matrix.with_qt6}}'
         with:
           timeout_minutes: 10
           retry_wait_seconds: 30
@@ -63,6 +74,7 @@ jobs:
             -DCMAKE_CXX_FLAGS='${{matrix.compiler_flags}}'
             -DCMAKE_C_FLAGS='${{matrix.compiler_flags}}'
             -DCMAKE_CXX_COMPILER='${{matrix.compiler}}'
+            -DWITH_QT6='${{matrix.with_qt6}}'
 
       - name: Install with CMake
         run: cmake --install '${{runner.workspace}}/build'

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,14 +98,12 @@ if (PEDANTIC)
             -Wlogical-op
             -Wnoexcept
             -Wstrict-null-sentinel
+            -Wno-unknown-pragmas
             )
     else()
         list(APPEND copyq_pedantic_flags
             -Weverything
             -Winconsistent-missing-override
-            )
-
-        list(APPEND copyq_pedantic_flags
             -Wno-c++98-compat
             -Wno-c++98-compat-pedantic
             -Wno-disabled-macro-expansion

--- a/plugins/itemencrypted/CMakeLists.txt
+++ b/plugins/itemencrypted/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(copyq_plugin_itemencrypted_SOURCES
+    ../../src/common/command.cpp
     ../../src/common/config.cpp
     ../../src/common/log.cpp
     ../../src/common/mimetypes.cpp

--- a/plugins/itempinned/CMakeLists.txt
+++ b/plugins/itempinned/CMakeLists.txt
@@ -1,6 +1,7 @@
 set(copyq_plugin_itempinned_SOURCES
     ../../src/item/itemwidgetwrapper.cpp
     ../../src/item/itemsaverwrapper.cpp
+    ../../src/common/command.cpp
     ../../src/common/display.cpp
     ../../src/gui/screen.cpp
     )

--- a/plugins/itemtags/CMakeLists.txt
+++ b/plugins/itemtags/CMakeLists.txt
@@ -1,6 +1,7 @@
 set(copyq_plugin_itemtags_SOURCES
     ../../src/item/itemwidgetwrapper.cpp
     ../../src/item/itemsaverwrapper.cpp
+    ../../src/common/command.cpp
     ../../src/common/config.cpp
     ../../src/common/log.cpp
     ../../src/common/mimetypes.cpp

--- a/src/gui/mainwindow.cpp
+++ b/src/gui/mainwindow.cpp
@@ -716,8 +716,14 @@ void MainWindow::exit()
 
 void MainWindow::closeEvent(QCloseEvent *event)
 {
-    event->ignore();
     hideWindow();
+#if QT_VERSION >= QT_VERSION_CHECK(5,15,0)
+    event->accept();
+#else
+    // This is needed in older versions of Qt (tested in Qt 5.12.8),
+    // otherwise focusing is somewhat broken (test pasteFromMainWindow()).
+    event->ignore();
+#endif
     COPYQ_LOG("Got main window close event.");
 }
 

--- a/src/gui/theme.cpp
+++ b/src/gui/theme.cpp
@@ -146,35 +146,22 @@ int fitFontWeight(int weight, int low, int high, int highCss)
 
 int getFontWeightForStyleSheet(int weight)
 {
-    // Copied from QFont sources (not available in all Qt versions).
-    enum Weight {
-        Thin     = 0,    // 100
-        ExtraLight = 12, // 200
-        Light    = 25,   // 300
-        Normal   = 50,   // 400
-        Medium   = 57,   // 500
-        DemiBold = 63,   // 600
-        Bold     = 75,   // 700
-        ExtraBold = 81,  // 800
-        Black    = 87    // 900
-    };
-
-    if (weight < ExtraLight)
-        return fitFontWeight(weight, Thin, ExtraLight, 200);
-    if (weight < Light)
-        return fitFontWeight(weight, ExtraLight, Light, 300);
-    if (weight < Normal)
-        return fitFontWeight(weight, Light, Normal, 400);
-    if (weight < Medium)
-        return fitFontWeight(weight, Normal, Medium, 500);
-    if (weight < DemiBold)
-        return fitFontWeight(weight, Medium, DemiBold, 600);
-    if (weight < Bold)
-        return fitFontWeight(weight, DemiBold, Bold, 700);
-    if (weight < ExtraBold)
-        return fitFontWeight(weight, Bold, ExtraBold, 800);
-    if (weight < Black)
-        return fitFontWeight(weight, ExtraBold, Black, 900);
+    if (weight < QFont::ExtraLight)
+        return fitFontWeight(weight, QFont::Thin, QFont::ExtraLight, 200);
+    if (weight < QFont::Light)
+        return fitFontWeight(weight, QFont::ExtraLight, QFont::Light, 300);
+    if (weight < QFont::Normal)
+        return fitFontWeight(weight, QFont::Light, QFont::Normal, 400);
+    if (weight < QFont::Medium)
+        return fitFontWeight(weight, QFont::Normal, QFont::Medium, 500);
+    if (weight < QFont::DemiBold)
+        return fitFontWeight(weight, QFont::Medium, QFont::DemiBold, 600);
+    if (weight < QFont::Bold)
+        return fitFontWeight(weight, QFont::DemiBold, QFont::Bold, 700);
+    if (weight < QFont::ExtraBold)
+        return fitFontWeight(weight, QFont::Bold, QFont::ExtraBold, 800);
+    if (weight < QFont::Black)
+        return fitFontWeight(weight, QFont::ExtraBold, QFont::Black, 900);
 
     return 900;
 }

--- a/src/scriptable/scriptable.cpp
+++ b/src/scriptable/scriptable.cpp
@@ -2004,13 +2004,13 @@ QJSValue Scriptable::setSelectedItemData()
 
 QJSValue Scriptable::selectedItemsData()
 {
-    return toScriptValue( m_proxy->selectedItemsData(), this );
+    return toScriptValue( m_proxy->selectedItemsData().items, this );
 }
 
 void Scriptable::setSelectedItemsData()
 {
     m_skipArguments = 1;
-    const auto dataList = fromScriptValue<QVector<QVariantMap>>( argument(0), this );
+    const VariantMapList dataList{fromScriptValue<QVector<QVariantMap>>( argument(0), this )};
     m_proxy->setSelectedItemsData(dataList);
 }
 
@@ -2207,18 +2207,18 @@ QJSValue Scriptable::menuItems()
     const auto text = argument(0);
     if ( text.isString() ) {
         m_skipArguments = -1;
-        QVector<QVariantMap> items;
+        VariantMapList items;
         for (const auto &arg : arguments())
-            items.append(createDataMap(mimeText, arg));
+            items.items.append(createDataMap(mimeText, arg));
         const int i = m_proxy->menuItems(items);
-        if (i == -1 || i >= items.size())
+        if (i == -1 || i >= items.items.size())
             return QString();
-        return getTextData(items[i]);
+        return getTextData(items.items[i]);
     }
 
     m_skipArguments = 1;
-    const auto items = fromScriptValue<QVector<QVariantMap>>(text, this);
-    if ( items.isEmpty() )
+    const VariantMapList items{fromScriptValue<QVector<QVariantMap>>(text, this)};
+    if ( items.items.isEmpty() )
         return -1;
     return m_proxy->menuItems(items);
 }
@@ -3189,7 +3189,7 @@ QJSValue Scriptable::changeItem(bool create)
     }
 
     QString error;
-    const QVector<QVariantMap> items = getItemArguments(i, args, &error);
+    const VariantMapList items{getItemArguments(i, args, &error)};
     if ( !error.isEmpty() )
         return throwError(error);
 
@@ -3493,7 +3493,7 @@ void Scriptable::insert(int row, int argumentsBegin, int argumentsEnd)
 {
     m_skipArguments = argumentsEnd;
 
-    const QVector<QVariantMap> items = getItemList(argumentsBegin, argumentsEnd, argumentsArray());
+    const VariantMapList items{getItemList(argumentsBegin, argumentsEnd, argumentsArray())};
     const auto error = m_proxy->browserInsert(m_tabName, row, items);
     if ( !error.isEmpty() )
         throwError(error);

--- a/src/scriptable/scriptable.cpp
+++ b/src/scriptable/scriptable.cpp
@@ -616,8 +616,11 @@ QJSValue fromUnicode(const QString &text, const QJSValue &codecName, Scriptable 
     const auto encoding = encodingFromNameOrThrow(codecName, scriptable);
     if (!encoding)
         return QJSValue();
+    const QStringConverter::Flags flags = *encoding == QStringConverter::Utf8
+        ? QStringConverter::Flag::Default
+        : (QStringConverter::Flag::Default | QStringConverter::Flag::WriteBom);
     return scriptable->newByteArray(
-        QStringEncoder(*encoding).encode(text) );
+        QStringEncoder(*encoding, flags).encode(text) );
 }
 #endif
 

--- a/src/scriptable/scriptableproxy.cpp
+++ b/src/scriptable/scriptableproxy.cpp
@@ -921,7 +921,7 @@ QByteArray ScriptableProxy::callFunctionHelper(const QByteArray &serializedFunct
         auto &value = arguments[i];
         const int argumentTypeId = metaMethod.parameterType(i);
         if (argumentTypeId == QMetaType::QVariant) {
-            args[i] = Q_ARG(QVariant, value);
+            args[i] = QGenericArgument( "QVariant", static_cast<void*>(value.data()) );
         } else if ( value.userType() == argumentTypeId ) {
             args[i] = QGenericArgument( value.typeName(), static_cast<void*>(value.data()) );
         } else {
@@ -949,8 +949,8 @@ QByteArray ScriptableProxy::callFunctionHelper(const QByteArray &serializedFunct
         returnValue = QVariant(typeId, nullptr);
 #endif
         const auto genericReturnValue = returnValue.isValid()
-                ? QGenericReturnArgument(returnValue.typeName(), static_cast<void*>(returnValue.data()) )
-                : Q_RETURN_ARG(QVariant, returnValue);
+                ? QGenericReturnArgument( returnValue.typeName(), static_cast<void*>(returnValue.data()) )
+                : QGenericReturnArgument( "QVariant", static_cast<void*>(returnValue.data()) );
 
         called = metaMethod.invoke(
                 this, genericReturnValue,

--- a/src/scriptable/scriptableproxy.cpp
+++ b/src/scriptable/scriptableproxy.cpp
@@ -91,7 +91,7 @@ void registerMetaTypes() {
     qRegisterMetaTypeStreamOperators<ScriptablePath>("ScriptablePath");
     qRegisterMetaTypeStreamOperators<QVector<int>>("QVector<int>");
     qRegisterMetaTypeStreamOperators<QVector<Command>>("QVector<Command>");
-    qRegisterMetaTypeStreamOperators<QVector<QVariantMap>>("QVector<QVariantMap>");
+    qRegisterMetaTypeStreamOperators<VariantMapList>("VariantMapList");
     qRegisterMetaTypeStreamOperators<Qt::KeyboardModifiers>("Qt::KeyboardModifiers");
 #else
     qRegisterMetaType<QPointer<QWidget>>("QPointer<QWidget>");
@@ -102,7 +102,7 @@ void registerMetaTypes() {
     qRegisterMetaType<ScriptablePath>("ScriptablePath");
     qRegisterMetaType<QVector<int>>("QVector<int>");
     qRegisterMetaType<QVector<Command>>("QVector<Command>");
-    qRegisterMetaType<QVector<QVariantMap>>("QVector<QVariantMap>");
+    qRegisterMetaType<VariantMapList>("VariantMapList");
     qRegisterMetaType<Qt::KeyboardModifiers>("Qt::KeyboardModifiers");
 #endif
 
@@ -225,6 +225,20 @@ QDataStream &operator>>(QDataStream &in, NamedValueList &list)
         in >> item.name >> item.value;
         list.append(item);
     }
+    Q_ASSERT(in.status() == QDataStream::Ok);
+    return in;
+}
+
+QDataStream &operator<<(QDataStream &out, const VariantMapList &items)
+{
+    out << items.items;
+    Q_ASSERT(out.status() == QDataStream::Ok);
+    return out;
+}
+
+QDataStream &operator>>(QDataStream &in, VariantMapList &items)
+{
+    in >> items.items;
     Q_ASSERT(in.status() == QDataStream::Ok);
     return in;
 }
@@ -1397,7 +1411,7 @@ int ScriptableProxy::findTabIndex(const QString &arg1)
     return m_wnd->findTabIndex(arg1);
 }
 
-int ScriptableProxy::menuItems(const QVector<QVariantMap> &items)
+int ScriptableProxy::menuItems(const VariantMapList &items)
 {
     INVOKE(menuItems, (items));
 
@@ -1407,7 +1421,7 @@ int ScriptableProxy::menuItems(const QVector<QVariantMap> &items)
 
     const auto addMenuItems = [&](const QString &searchText) {
         menu.clearClipboardItems();
-        for (const QVariantMap &data : items) {
+        for (const QVariantMap &data : items.items) {
             const QString text = getTextData(data);
             if ( text.contains(searchText, Qt::CaseInsensitive) )
                 menu.addClipboardItemAction(data, true);
@@ -1422,7 +1436,7 @@ int ScriptableProxy::menuItems(const QVector<QVariantMap> &items)
     if (act == nullptr)
         return -1;
 
-    return items.indexOf(act->data().toMap());
+    return items.items.indexOf(act->data().toMap());
 }
 
 void ScriptableProxy::openActionDialog(const QVariantMap &arg1)
@@ -1505,7 +1519,7 @@ bool ScriptableProxy::browserOpenEditor(const QString &tabName, const QByteArray
     return c && c->openEditor(arg1, changeClipboard);
 }
 
-QString ScriptableProxy::browserInsert(const QString &tabName, int row, const QVector<QVariantMap> &items)
+QString ScriptableProxy::browserInsert(const QString &tabName, int row, const VariantMapList &items)
 {
     INVOKE(browserInsert, (tabName, row, items));
 
@@ -1513,10 +1527,10 @@ QString ScriptableProxy::browserInsert(const QString &tabName, int row, const QV
     if (!c)
         return QLatin1String("Invalid tab");
 
-    if ( !c->allocateSpaceForNewItems(items.size()) )
+    if ( !c->allocateSpaceForNewItems(items.items.size()) )
         return QLatin1String("Tab is full (cannot remove any items)");
 
-    for (const auto &item : items) {
+    for (const auto &item : items.items) {
         if ( !c->add(item, row) )
             return QLatin1String("Failed to new add items");
     }
@@ -1524,7 +1538,7 @@ QString ScriptableProxy::browserInsert(const QString &tabName, int row, const QV
     return QString();
 }
 
-QString ScriptableProxy::browserChange(const QString &tabName, int row, const QVector<QVariantMap> &items)
+QString ScriptableProxy::browserChange(const QString &tabName, int row, const VariantMapList &items)
 {
     INVOKE(browserChange, (tabName, row, items));
 
@@ -1533,7 +1547,7 @@ QString ScriptableProxy::browserChange(const QString &tabName, int row, const QV
         return QLatin1String("Invalid tab");
 
     int currentRow = row;
-    for (const auto &data : items) {
+    for (const auto &data : items.items) {
         const auto index = c->index(currentRow);
         QVariantMap itemData = c->model()->data(index, contentType::data).toMap();
         for (auto it = data.constBegin(); it != data.constEnd(); ++it) {
@@ -1649,13 +1663,13 @@ bool ScriptableProxy::setSelectedItemData(int selectedIndex, const QVariantMap &
     return c->model()->setData(index, data, contentType::data);
 }
 
-QVector<QVariantMap> ScriptableProxy::selectedItemsData()
+VariantMapList ScriptableProxy::selectedItemsData()
 {
     INVOKE(selectedItemsData, ());
 
     auto c = currentBrowser();
     if (!c)
-        return QVector<QVariantMap>();
+        return {};
 
     const auto model = c->model();
 
@@ -1670,10 +1684,10 @@ QVector<QVariantMap> ScriptableProxy::selectedItemsData()
         }
     }
 
-    return dataList;
+    return {dataList};
 }
 
-void ScriptableProxy::setSelectedItemsData(const QVector<QVariantMap> &dataList)
+void ScriptableProxy::setSelectedItemsData(const VariantMapList &dataList)
 {
     INVOKE2(setSelectedItemsData, (dataList));
 
@@ -1684,12 +1698,12 @@ void ScriptableProxy::setSelectedItemsData(const QVector<QVariantMap> &dataList)
     const auto model = c->model();
 
     const auto indexes = selectedIndexes();
-    const auto count = std::min( indexes.size(), dataList.size() );
+    const auto count = std::min( indexes.size(), dataList.items.size() );
     for ( int i = 0; i < count; ++i ) {
         const auto &index = indexes[i];
         if ( index.isValid() ) {
             Q_ASSERT( index.model() == model );
-            model->setData(index, dataList[i], contentType::data);
+            model->setData(index, dataList.items[i], contentType::data);
         }
     }
 }

--- a/src/scriptable/scriptableproxy.cpp
+++ b/src/scriptable/scriptableproxy.cpp
@@ -212,8 +212,9 @@ QDataStream &operator<<(QDataStream &out, const NamedValueList &list)
 
 QDataStream &operator>>(QDataStream &in, NamedValueList &list)
 {
-    int size;
+    decltype(list.size()) size;
     in >> size;
+    list.reserve(size);
     for (int i = 0; i < size; ++i) {
         NamedValue item;
         in >> item.name >> item.value;

--- a/src/scriptable/scriptableproxy.cpp
+++ b/src/scriptable/scriptableproxy.cpp
@@ -183,20 +183,25 @@ void selectionRemoveInvalid(QList<QPersistentModelIndex> *indexes)
 
 Q_DECLARE_METATYPE(QFile*)
 
-QDataStream &operator<<(QDataStream &out, const NotificationButton &button)
+QDataStream &operator<<(QDataStream &out, const NotificationButtons &list)
 {
-    out << button.name
-        << button.script
-        << button.data;
+    out << list.size();
+    for (const auto &button : list)
+        out << button.name << button.script << button.data;
     Q_ASSERT(out.status() == QDataStream::Ok);
     return out;
 }
 
-QDataStream &operator>>(QDataStream &in, NotificationButton &button)
+QDataStream &operator>>(QDataStream &in, NotificationButtons &list)
 {
-    in >> button.name
-       >> button.script
-       >> button.data;
+    decltype(list.size()) size;
+    in >> size;
+    list.reserve(size);
+    for (int i = 0; i < size; ++i) {
+        NotificationButton button;
+        in >> button.name >> button.script >> button.data;
+        list.append(button);
+    }
     Q_ASSERT(in.status() == QDataStream::Ok);
     return in;
 }

--- a/src/scriptable/scriptableproxy.h
+++ b/src/scriptable/scriptableproxy.h
@@ -33,6 +33,10 @@ struct NamedValue {
     QVariant value;
 };
 
+struct VariantMapList {
+    QVector<QVariantMap> items;
+};
+
 using NamedValueList = QVector<NamedValue>;
 
 struct ScriptablePath {
@@ -47,7 +51,7 @@ struct ItemSelection {
 Q_DECLARE_METATYPE(NamedValueList)
 Q_DECLARE_METATYPE(ScriptablePath)
 Q_DECLARE_METATYPE(NotificationButtons)
-Q_DECLARE_METATYPE(QVector<QVariantMap>)
+Q_DECLARE_METATYPE(VariantMapList)
 Q_DECLARE_METATYPE(Qt::KeyboardModifiers)
 
 #if QT_VERSION < QT_VERSION_CHECK(6,0,0)
@@ -58,6 +62,8 @@ QDataStream &operator<<(QDataStream &out, const NotificationButtons &list);
 QDataStream &operator>>(QDataStream &in, NotificationButtons &list);
 QDataStream &operator<<(QDataStream &out, const NamedValueList &list);
 QDataStream &operator>>(QDataStream &in, NamedValueList &list);
+QDataStream &operator<<(QDataStream &out, const VariantMapList &list);
+QDataStream &operator>>(QDataStream &in, VariantMapList &list);
 QDataStream &operator<<(QDataStream &out, ClipboardMode mode);
 QDataStream &operator>>(QDataStream &in, ClipboardMode &mode);
 QDataStream &operator<<(QDataStream &out, const ScriptablePath &path);
@@ -142,7 +148,7 @@ public slots:
     bool toggleCurrentMenu();
     int findTabIndex(const QString &arg1);
 
-    int menuItems(const QVector<QVariantMap> &items);
+    int menuItems(const VariantMapList &items);
 
     void openActionDialog(const QVariantMap &arg1);
 
@@ -159,8 +165,8 @@ public slots:
     int browserLength(const QString &tabName);
     bool browserOpenEditor(const QString &tabName, const QByteArray &arg1, bool changeClipboard);
 
-    QString browserInsert(const QString &tabName, int row, const QVector<QVariantMap> &items);
-    QString browserChange(const QString &tabName, int row, const QVector<QVariantMap> &items);
+    QString browserInsert(const QString &tabName, int row, const VariantMapList &items);
+    QString browserChange(const QString &tabName, int row, const VariantMapList &items);
 
     QByteArray browserItemData(const QString &tabName, int arg1, const QString &arg2);
     QVariantMap browserItemData(const QString &tabName, int arg1);
@@ -177,8 +183,8 @@ public slots:
     QVariantMap selectedItemData(int selectedIndex);
     bool setSelectedItemData(int selectedIndex, const QVariantMap &data);
 
-    QVector<QVariantMap> selectedItemsData();
-    void setSelectedItemsData(const QVector<QVariantMap> &dataList);
+    VariantMapList selectedItemsData();
+    void setSelectedItemsData(const VariantMapList &dataList);
 
     int createSelection(const QString &tabName);
     int selectionCopy(int id);

--- a/src/scriptable/scriptableproxy.h
+++ b/src/scriptable/scriptableproxy.h
@@ -54,8 +54,8 @@ Q_DECLARE_METATYPE(Qt::KeyboardModifiers)
 Q_DECLARE_METATYPE(ClipboardMode)
 #endif
 
-QDataStream &operator<<(QDataStream &out, const NotificationButton &button);
-QDataStream &operator>>(QDataStream &in, NotificationButton &button);
+QDataStream &operator<<(QDataStream &out, const NotificationButtons &list);
+QDataStream &operator>>(QDataStream &in, NotificationButtons &list);
 QDataStream &operator<<(QDataStream &out, const NamedValueList &list);
 QDataStream &operator>>(QDataStream &in, NamedValueList &list);
 QDataStream &operator<<(QDataStream &out, ClipboardMode mode);

--- a/utils/github/install-linux.sh
+++ b/utils/github/install-linux.sh
@@ -2,33 +2,60 @@
 # Installs build dependencies.
 set -xeuo pipefail
 
+if [[ $WITH_QT6 == true ]]; then
+    qt_packages=(
+        libegl-dev
+
+        qt6-base-private-dev
+        qt6-base-dev
+        qt6-base-dev-tools
+        qt6-tools-dev
+        qt6-tools-dev-tools
+        qt6-l10n-tools
+        qt6-declarative-dev
+
+        libqt6svg6-dev
+        libqt6svg6
+
+        # Wayland support
+        libqt6waylandclient6
+        qt6-wayland
+        qt6-wayland-dev
+        qt6-wayland-dev-tools
+    )
+else
+    qt_packages=(
+        qt5-default
+
+        qtbase5-private-dev
+        qtdeclarative5-dev
+        qttools5-dev
+        qttools5-dev-tools
+
+        libqt5x11extras5-dev
+
+        libqt5svg5-dev
+        libqt5svg5
+
+        # Wayland support
+        libqt5waylandclient5-dev
+        qtwayland5
+        qtwayland5-dev-tools
+
+        # KDE Frameworks
+        libkf5notifications-dev
+        extra-cmake-modules
+    )
+fi
+
 packages=(
-    qt5-default
-
-    qtbase5-private-dev
-    qtdeclarative5-dev
-    qttools5-dev
-    qttools5-dev-tools
-
-    libqt5x11extras5-dev
-
-    libqt5svg5-dev
-    libqt5svg5
-
     # fixes some issues on X11
     libxfixes-dev
     libxtst-dev
 
     # Wayland support
     extra-cmake-modules
-    libqt5waylandclient5-dev
     libwayland-dev
-    qtwayland5
-    qtwayland5-dev-tools
-
-    # KDE Frameworks
-    libkf5notifications-dev
-    extra-cmake-modules
 
     # encryption plugin
     gnupg2
@@ -39,4 +66,4 @@ packages=(
 )
 
 sudo apt-get update
-sudo apt-get install "${packages[@]}" "$@"
+sudo apt-get install "${qt_packages[@]}" "${packages[@]}" "$@"

--- a/utils/github/test-linux.sh
+++ b/utils/github/test-linux.sh
@@ -22,4 +22,4 @@ rm -rf ~/.config/copyq.test
 
 # Run tests.
 export COPYQ_TESTS_RERUN_FAILED=0
-./copyq tests
+./copyq tests "$@"

